core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -238,12 +238,8 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
 
                 log.debug("Updated the epoch cache from remote tier till offset: {} with size: {} for {}", leaderLocalLogStartOffset, epochs.size(), partition);
 
-                // Restore producer snapshot
-                File snapshotFile = LogFileUtils.producerSnapshotFile(unifiedLog.dir(), nextOffset);
-                buildProducerSnapshotFile(snapshotFile, remoteLogSegmentMetadata, rlm);
-
                 // Reload producer snapshots.
-                unifiedLog.producerStateManager().truncateFullyAndReloadSnapshots();
+                truncateFullyAndReloadRestoredSnapshots(unifiedLog, nextOffset, remoteLogSegmentMetadata, rlm);
```

Comment:
Nice catch @hudeqi! This is the expected behaviour, not sure why this was changed while porting it to trunk:

https://sourcegraph.com/github.com/satishd/kafka@2.8.x-tiered-storage/-/blob/core/src/main/scala/kafka/server/ReplicaFetcherThread.scala?L422-437